adage/analyze/api.py

```diff
@@ -427,7 +427,14 @@ def apply_filters(self, request, applicable_filters):
                 raise BadRequest("Invalid gene IDs: %s" % genes)
             # "in" operator in Django supports both list and set.
             qset = Q(gene1__in=ids) | Q(gene2__in=ids)
-            object_list = object_list.filter(qset).distinct()
+            direct_edges = Edge.objects.filter(qset).distinct()
+            related_genes = set()
+            for e in direct_edges:
+                related_genes.add(e.gene1)
+                related_genes.add(e.gene2)
+            object_list = object_list.filter(
+                gene1__in=related_genes, gene2__in=related_genes
+            ).distinct()
```

Comment:
Let me see if I understand. The user request was that we add edges between the genes that were returned - not just the query genes. There are two ways to interpret this:

1. Query the gene (dnaA) -> This provides a list of related genes (up to X) -> query for all edges between any genes in X + dnaA -> Display any edges (+associated nodes) over the slider cutoff.

2. Query the gene (dnaA) -> This provides a list of related genes (up to X) -> query for all edges between any genes in X + dnaA -> Display any edges between genes that have an edge that is over the slider-cutoff with dnaA.

Did I get that framing correct?